lib/puppet/provider/package/pecl.rb

```diff
@@ -20,19 +20,17 @@ def self.pecllist(hash)
     begin
       list = execute(command).split("\n").map do |set|
         if hash[:justme]
-          if %r{^#{hash[:justme]}$}i.match(set)
+          if %r{^#{hash[:justme]}$}i =~ set
             if peclhash = peclsplit(set) # rubocop:disable Lint/AssignmentInCondition
```

Comment:
The two versions should be functionally equivalent, as a non-branching if returns nil in Ruby.

I guess the real question is if this:

```ruby
if %r{^#{hash[:justme]}$}i =~ set
  if peclhash = peclsplit(set) # rubocop:disable Lint/AssignmentInCondition
    peclhash[:provider] = :peclcmd
    peclhash
  end
end
```
is more readable than:

```ruby
if %r{^#{hash[:justme]}$}i =~ set &&
   peclhash = peclsplit(set) # rubocop:disable Lint/AssignmentInCondition
  peclhash[:provider] = :peclcmd
  peclhash
end
```